src/display/text_layer.js

```diff
@@ -638,6 +638,22 @@ const renderTextLayer = (function renderTextLayerClosure() {
 
     _processItems(items, styleCache) {
       for (let i = 0, len = items.length; i < len; i++) {
+        if (!("str" in items[i])) {
```

Comment:
I seem to recall that `in` can perhaps be less efficient than just checking the value, so can't we just do the following instead?
```suggestion
        if (items[i].str === undefined) {
```